<dependencies>
    <!-- SLF4J API -->
    <dependency>
        <groupId>org.slf4j</groupId>
        <artifactId>slf4j-api</artifactId>
        <version>2.0.13</version>
    </dependency>

    <!-- Logback Classic (backend for SLF4J) -->
    <dependency>
        <groupId>ch.qos.logback</groupId>
        <artifactId>logback-classic</artifactId>
        <version>1.4.14</version>
    </dependency>
</dependencies>



import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class Application {
    private static final Logger logger = LoggerFactory.getLogger(Application.class);

    public static void main(String[] args) {
        int x = 10;
        int y = 0;

        logger.warn("This is a warning: the value of y is {}", y);

        try {
            int result = x / y;
        } catch (ArithmeticException e) {
            logger.error("An error occurred while dividing {} by {}: {}", x, y, e.toString());
        }
    }
}


OUTPUT:

WARN  Application - This is a warning: the value of y is 0
ERROR Application - An error occurred while dividing 10 by 0: java.lang.ArithmeticException: / by zero